### PR TITLE
Adds oriented contraint check to Homography RANSAC

### PR DESCRIPTION
**Pull request by ducha-aiki**
#### Changes
Removes bad samples prior to model estimation to speed-up RANSAC.
References: 
http://www.dia.fi.upm.es/~pcr/publications/jrtip2015.pdf - paper, formula (5)
OpenCV implementation: https://github.com/opencv/opencv/blob/4.x/modules/calib3d/src/usac/degeneracy.cpp#L88

#### Type of change

- [ ] 📚  Documentation Update
- [ ] 🧪 Tests Cases
- [ ] 🐞 Bug fix (non-breaking change which fixes an issue)
- [x] 🔬 New feature (non-breaking change which adds functionality)
- [ ] 🚨 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📝 This change requires a documentation update

#### Checklist

- [ ] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] Did you update CHANGELOG in case of a major change?

### CODE REVIEW

File: kornia/geometry/homography.py [hunk 1]
```diff
@@ -180,3 +180,37 @@ def find_homography_dlt_iterated(
         weights_new: torch.Tensor = torch.exp(-errors / (2.0 * (soft_inl_th ** 2)))
         H = find_homography_dlt(points1, points2, weights_new)
     return H
+
+
+def sample_is_valid_for_homography(points1: torch.Tensor, points2: torch.Tensor) -> torch.Tensor:
+    '''Function, which implements oriented contraint check from :cite:`Marquez-Neila2015`
```
Comment (edgarriba) [inline kornia/geometry/homography.py:186]:
```suggestion
    """Function, which implements oriented constraint check from :cite:`Marquez-Neila2015`.
```

File: kornia/geometry/homography.py [hunk 2]
```diff
@@ -180,3 +180,37 @@ def find_homography_dlt_iterated(
...
+    Analogous to https://github.com/opencv/opencv/blob/4.x/modules/calib3d/src/usac/degeneracy.cpp#L88
```
Comment (edgarriba) [inline kornia/geometry/homography.py:187]:
```suggestion

    Analogous to https://github.com/opencv/opencv/blob/4.x/modules/calib3d/src/usac/degeneracy.cpp#L88
```

File: kornia/geometry/homography.py [hunk 3]
```diff
@@ -180,3 +180,37 @@ def find_homography_dlt_iterated(
...
+    Args:
```
Comment (edgarriba) [inline kornia/geometry/homography.py:188]:
```suggestion

    Args:
```

File: kornia/geometry/homography.py [hunk 4]
```diff
@@ -180,3 +180,37 @@ def find_homography_dlt_iterated(
...
+        points1: A set of points in the first image with a tensor shape :math:`(B, 4, 2)`.
+        points2: A set of points in the second image with a tensor shape :math:`(B, 4, 2)`.
+
+    Returns:
+        mask: if the minimal sample is good for homography estimation:math:`(B, 3, 3)`.
```
Comment (edgarriba) [inline kornia/geometry/homography.py:193]:
```suggestion
        Mask with the minimal sample is good for homography estimation:math:`(B, 3, 3)`.
```

File: kornia/geometry/homography.py [hunk 5]
```diff
@@ -180,3 +180,37 @@ def find_homography_dlt_iterated(
...
+        '''
```
Comment (edgarriba) [inline kornia/geometry/homography.py:194]:
```suggestion
    """
```

**Review by edgarriba — APPROVED** [review]
LGTM. Just docs things

**Comment by ducha-aiki** [comment]
Some timings (here is colab https://colab.research.google.com/drive/11vyYaL1yEnofFXjZNR5d40VtYVkZq3ZF?usp=sharing)

## GPU Colab
Original kornia RANSAC: 629 ms 
 + https://github.com/kornia/kornia/pull/1454: 502 ms
+ https://github.com/kornia/kornia/pull/1454 + this PR:   455 ms

## CPU Colab:

Original kornia RANSAC: 2.14 s
 + https://github.com/kornia/kornia/pull/1454:   1.25 s
+ https://github.com/kornia/kornia/pull/1454 + this PR:   1.02 s

**Pull request merged**